main/src/mill/main/MainModule.scala

```diff
@@ -173,4 +175,17 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets will clean everything.
+    */
+  def clean(targets: String*) = mill.T.command {
+    val result = targets match {
+      case Nil =>
+        List(ammonite.ops.%%('rm, "-rf", OutDir)(ammonite.ops.pwd).out.string.trim())
```

Comment:
probably should use `ammonite.ops.rm` here

Reply:
Aw okay didn't even know that was a thing.

Reply:
Good point. I did note that other functions in this file do the following:

```
    val resolved = RunScript.resolveTasks(
      mill.main.ResolveMetadata, evaluator, targets, multiSelect = true
    )
```
And then I extract the names from `resolved`. Would that be sufficient? I will play with it later.

Reply:
would it not be enough to validate that what you pass to rm -rf contains `out`?

Reply:
And maybe that the parent folder contains build.sc?

Reply:
Yes you're right. I guess I'm wondering if there is also value is validating that the passed targets are indeed valid targets.

Reply:
You probably want to convert the path to a absolute path first before applying tests.